lib/src/screens/buy/buy_sell_page.dart

```diff
@@ -0,0 +1,475 @@
+
+    _onCurrencyChange(buySellViewModel.cryptoCurrency, buySellViewModel, cryptoCurrencyKey);
+    _onCurrencyChange(buySellViewModel.fiatCurrency, buySellViewModel, fiatCurrencyKey);
+
+    reaction(
+            (_) => buySellViewModel.wallet.name,
+            (String _) =>
+            _onWalletNameChange(buySellViewModel, buySellViewModel.cryptoCurrency, cryptoCurrencyKey));
+
+    reaction(
+        (_) => buySellViewModel.cryptoCurrency,
+        (CryptoCurrency currency) =>
+            _onCurrencyChange(currency, buySellViewModel, cryptoCurrencyKey));
+
+    reaction(
+        (_) => buySellViewModel.fiatCurrency,
+        (FiatCurrency currency) =>
+            _onCurrencyChange(currency, buySellViewModel, fiatCurrencyKey));
+
+    reaction((_) => buySellViewModel.fiatAmount, (String amount) {
+      if (fiatCurrencyKey.currentState!.amountController.text != amount) {
+        fiatCurrencyKey.currentState!.amountController.text = amount;
+      }
+    });
+
+    reaction((_) => buySellViewModel.isCryptoCurrencyAddressEnabled, (bool isEnabled) {
+      cryptoCurrencyKey.currentState!.isAddressEditable(isEditable: isEnabled);
+    });
+
+    reaction((_) => buySellViewModel.cryptoAmount, (String amount) {
+      if (cryptoCurrencyKey.currentState!.amountController.text != amount) {
+        cryptoCurrencyKey.currentState!.amountController.text = amount;
+      }
+    });
+
+    reaction((_) => buySellViewModel.cryptoCurrencyAddress, (String address) {
+      if (cryptoAddressController != address) {
+        cryptoCurrencyKey.currentState!.addressController.text = address;
+      }
+    });
+
+    fiatAmountController.addListener(() {
+      if (fiatAmountController.text != buySellViewModel.fiatAmount) {
+          buySellViewModel.changeFiatAmount(amount: fiatAmountController.text);
+      }
+    });
+
+    cryptoAmountController.addListener(() {
+      if (cryptoAmountController.text != buySellViewModel.cryptoAmount) {
+          buySellViewModel.changeCryptoAmount(amount: cryptoAmountController.text);
+      }
+    });
+
+    cryptoAddressController.addListener(() {
+      buySellViewModel.changeCryptoCurrencyAddress(cryptoAddressController.text);
+    });
+
+    _cryptoAddressFocus.addListener(() async {
+      if (!_cryptoAddressFocus.hasFocus && cryptoAddressController.text.isNotEmpty) {
+        final domain = cryptoAddressController.text;
+        buySellViewModel.cryptoCurrencyAddress =
+        await fetchParsedAddress(context, domain, buySellViewModel.cryptoCurrency);
+      }
+    });
+
+    reaction((_) => buySellViewModel.wallet.walletAddresses.addressForExchange, (String address) {
+      if (buySellViewModel.cryptoCurrency == CryptoCurrency.xmr) {
+        cryptoCurrencyKey.currentState!.changeAddress(address: address);
+      }
+
+      if (buySellViewModel.cryptoCurrency == CryptoCurrency.xmr) {
+        cryptoCurrencyKey.currentState!.changeAddress(address: address);
+      }
```

Comment:
duplicate? intended?
```suggestion
      if (buySellViewModel.cryptoCurrency == CryptoCurrency.xmr) {
        cryptoCurrencyKey.currentState!.changeAddress(address: address);
      }
```

Reply:
fixed